go/mysql/conn.go

```diff
@@ -758,383 +758,430 @@ func (c *Conn) writeEOFPacket(flags uint16, warnings uint16) error {
+		return c.stmtExecute(handler, data)
+	case ComStmtSendLongData:
+		return c.sendLongData(data)
+	case ComStmtClose:
+		stmtID, ok := c.parseComStmtClose(data)
 		c.recycleReadPacket()
 		if ok {
-			switch operation {
-			case 0:
-				c.Capabilities |= CapabilityClientMultiStatements
-			case 1:
-				c.Capabilities &^= CapabilityClientMultiStatements
-			default:
-				log.Errorf("Got unhandled packet (ComSetOption default) from client %v, returning error: %v", c.ConnectionID, data)
-				if err := c.writeErrorPacket(ERUnknownComError, SSUnknownComError, "error handling packet: %v", data); err != nil {
-					log.Errorf("Error writing error packet to client: %v", err)
-					return err
-				}
-			}
-			if err := c.writeEndResult(false, 0, 0, 0); err != nil {
-				log.Errorf("Error writeEndResult error %v ", err)
-				return err
-			}
-		} else {
-			log.Errorf("Got unhandled packet (ComSetOption else) from client %v, returning error: %v", c.ConnectionID, data)
-			if err := c.writeErrorPacket(ERUnknownComError, SSUnknownComError, "error handling packet: %v", data); err != nil {
-				log.Errorf("Error writing error packet to client: %v", err)
-				return err
-			}
+			delete(c.PrepareData, stmtID)
 		}
-	case ComPrepare:
-		query := c.parseComPrepare(data)
+	case ComStmtReset:
+		c.stmtReset(data)
+	case ComResetConnection:
+		// Clean up and reset the connection
 		c.recycleReadPacket()
+		handler.ComResetConnection(c)
+		// Reset prepared statements
+		c.PrepareData = make(map[uint32]*PrepareData)
+		err = c.writeOKPacket(0, 0, 0, 0)
+		if err != nil {
+			c.writeErrorPacketFromError(err)
+		}
 
-		var queries []string
-		if c.Capabilities&CapabilityClientMultiStatements != 0 {
-			queries, err = sqlparser.SplitStatementToPieces(query)
-			if err != nil {
-				log.Errorf("Conn %v: Error splitting query: %v", c, err)
-				if werr := c.writeErrorPacketFromError(err); werr != nil {
-					// If we can't even write the error, we're done.
-					log.Errorf("Conn %v: Error writing query error: %v", c, werr)
-					return werr
-				}
-			}
-		} else {
-			queries = []string{query}
+	default:
+		log.Errorf("Got unhandled packet (default) from %s, returning error: %v", c, data)
+		c.recycleReadPacket()
+		if err := c.writeErrorPacket(ERUnknownComError, SSUnknownComError, "command handling not implemented yet: %v", data[0]); err != nil {
+			log.Errorf("Error writing error packet to %s: %s", c, err)
+			return false
 		}
+	}
+
+	return true
+}
 
-		if len(queries) != 1 {
-			return fmt.Errorf("can not prepare multiple statements")
+func (c *Conn) stmtReset(data []byte) bool {
+	stmtID, ok := c.parseComStmtReset(data)
+	c.recycleReadPacket()
+	if !ok {
+		log.Error("Got unhandled packet from client %v, returning error: %v", c.ConnectionID, data)
+		if err := c.writeErrorPacket(ERUnknownComError, SSUnknownComError, "error handling packet: %v", data); err != nil {
+			log.Error("Error writing error packet to client: %v", err)
+			return false
 		}
```

Comment:
This pattern of (log.Error, writeErrorPacket, if error writing then log error and return false) happens all over the place in this file.

Maybe we should consider DRYing this up with something like:

```
if !c.logAndWriteError(ERUnknownComError, SSUnknownComError, "error handling packet: %v", data)
    return false
```

Where `logAndWriteError` would encapsulate all the above?